examples/search.rs

```diff
@@ -40,6 +41,20 @@ async fn main() -> Result<()> {
             ..Default::default()
         })
         .await?;
+     */
+
+    client
+        .create_collection(
+            &CreateCollectionBuilder::default()
```

Comment:
I see we take a reference here. That is just a side effect of our function signature here, isn't it?

We _can_ change them into `AsRef<CollectionConfig>` to also allow passing in owned types. But lets leave it now and explore that when we rewrite our functions with the builder pattern.

Reply:
If we're introducing breaking changes anyways, we could also take `Into<CreateCollection>` here and add implementations for `From<&mut CreateCollectionBulider>` and `From<&CreateCollection>` for CreateCollection so we don't even have to call .build(). But this is just sugar at this point.